pageserver/src/http/models.rs

```diff
@@ -103,12 +100,24 @@ impl TenantConfigRequest {
     }
 }
 
+/// A state of an entity: tenant/timeline/layer/etc. that is used to manage it.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub enum State {
```

Comment:
I'd still prefer to call it `TenantState`, `State` seems too general. Even if timelines and layers happen to have the same possible states, it feels like "false sharing". More likely, tenants, timelines and layers have slightly different states.